src/AssetRegistry/StandardERC4626SubRegistry.sol

```diff
@@ -0,0 +1,172 @@
+/** 
+    Created by Arcadia Finance
+    https://www.arcadia.finance
+
+    SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity >=0.4.22 <0.9.0;
+
+import "./AbstractSubRegistry.sol";
+import "../interfaces/IERC4626.sol";
+import "../interfaces/ISubRegistry.sol";
+import "../interfaces/IMainRegistry.sol";
+import {FixedPointMathLib} from "../utils/FixedPointMathLib.sol";
+
+/**
+ * @title Sub-registry for Standard ERC4626 tokens
+ * @author Arcadia Finance
+ * @notice The StandardERC4626Registry stores pricing logic and basic information for ERC4626 tokens for which the underlying assets have direct price feed.
+ * @dev No end-user should directly interact with the StandardERC4626Registry, only the Main-registry, Oracle-Hub or the contract owner */
+contract StandardERC4626SubRegistry is SubRegistry {
+    using FixedPointMathLib for uint256;
+
+    struct AssetInformation {
+        uint64 assetUnit;
+        address assetAddress;
+        address underlyingAssetAddress;
+        address[] underlyingAssetOracleAddresses;
+    }
+
+    mapping(address => AssetInformation) public assetToInformation;
+
+    /**
+     * @notice A Sub-Registry must always be initialised with the address of the Main-Registry and of the Oracle-Hub
+     * @param mainRegistry The address of the Main-registry
+     * @param oracleHub The address of the Oracle-Hub
+     */
+    constructor(address mainRegistry, address oracleHub)
+        SubRegistry(mainRegistry, oracleHub)
+    {}
+
+    /**
+     * @notice Adds a new asset to the ATokenSubRegistry, or overwrites an existing asset.
+     * @param assetAddress The contract address of the asset
+     * @param assetCreditRatings The List of Credit Ratings for the asset for the different BaseCurrencies.
+     * @dev The list of Credit Ratings should or be as long as the number of baseCurrencies added to the Main Registry,
+     *      or the list must have length 0. If the list has length zero, the credit ratings of the asset for all baseCurrencies is
+     *      is initiated as credit rating with index 0 by default (worst credit rating).
+     * @dev The assets are added/overwritten in the Main-Registry as well.
+     *      By overwriting existing assets, the contract owner can temper with the value of assets already used as collateral
+     *      (for instance by changing the oracleaddres to a fake price feed) and poses a security risk towards protocol users.
+     *      This risk can be mitigated by setting the boolean "assetsUpdatable" in the MainRegistry to false, after which
+     *      assets are no longer updatable.
+     * @dev Assets can't have more than 18 decimals.
+     */
+    function setAssetInformation(
+        address assetAddress,
+        uint256[] calldata assetCreditRatings
+    ) external onlyOwner {
+        address underlyingAddress = address(IERC4626(assetAddress).asset());
+        (
+            uint64 assetUnit,
```

Comment:
In theory, the decimals of the ERC4626 can differ from the decimals of the underlying ERC20, No idea why anyone would do this, but maybe good idea to check explicitly that 10**(ERC4626.decimals()) == assetUnit?

Reply:
require(10 ** IERC4626(assetAddress).decimals() == assetUnit, "SR: Decimals of asset and underlying don't match");

Reply:
Jep